pkg/stats/view.go

```diff
@@ -0,0 +1,296 @@
+	view.BandwidthData = make([]float64, maxSlots)
+
+	// Fourth Row
+	view.ErrLabel = widgets.NewParagraph()
+	view.ErrLabel.Title = "Errors"
+	view.ErrLabel.BorderStyle.Fg = ui.ColorRed
+	view.ErrChart = widgets.NewSparkline()
+	view.ErrChart.LineColor = view.ErrLabel.BorderStyle.Fg
+	view.ErrGroup = widgets.NewSparklineGroup(view.ErrChart)
+	view.ErrGroup.BorderStyle.Fg = view.ErrLabel.BorderStyle.Fg
+	view.ErrData = make([]float64, maxSlots)
+
+	view.HitLabel = widgets.NewParagraph()
+	view.HitLabel.Title = "Hit Ratio"
+	view.HitLabel.BorderStyle.Fg = ui.ColorGreen
+	view.HitChart = widgets.NewSparkline()
+	view.HitChart.LineColor = view.HitLabel.BorderStyle.Fg
+	view.HitGroup = widgets.NewSparklineGroup(view.HitChart)
+	view.HitGroup.BorderStyle.Fg = view.HitLabel.BorderStyle.Fg
+	view.HitData = make([]float64, maxSlots)
+
+	// Fifth Row
+	view.IoLabel = widgets.NewParagraph()
+	view.IoLabel.Title = "Image Optimizer"
+	view.IoLabel.BorderStyle.Fg = ui.ColorYellow
+	view.IoChart = widgets.NewSparkline()
+	view.IoChart.LineColor = view.IoLabel.BorderStyle.Fg
+	view.IoGroup = widgets.NewSparklineGroup(view.IoChart)
+	view.IoGroup.BorderStyle.Fg = view.IoLabel.BorderStyle.Fg
+	view.IoData = make([]float64, maxSlots)
+
+	view.LoggingLabel = widgets.NewParagraph()
+	view.LoggingLabel.Title = "Logs Sent"
+	view.LoggingLabel.BorderStyle.Fg = ui.ColorBlue
+	view.LoggingChart = widgets.NewSparkline()
+	view.LoggingChart.LineColor = view.LoggingLabel.BorderStyle.Fg
+	view.LoggingGroup = widgets.NewSparklineGroup(view.LoggingChart)
+	view.LoggingGroup.BorderStyle.Fg = view.LoggingLabel.BorderStyle.Fg
+	view.LoggingData = make([]float64, maxSlots)
+
+	return &view, nil
+}
+
+func (v *View) SetLayout() {
+	v.Grid.Set(
+		ui.NewRow(0.05, v.Header),
+		ui.NewRow(0.14,
+			ui.NewCol(1.0/7, v.Hits),
+			ui.NewCol(1.0/7, v.HitTime),
+			ui.NewCol(1.0/7, v.Misses),
+			ui.NewCol(1.0/7, v.HitRatio),
+			ui.NewCol(1.0/7, v.MissTime),
+			ui.NewCol(1.0/7, v.Requests),
+			ui.NewCol(1.0/7, v.Errors),
+		),
+		ui.NewRow(0.15,
+			ui.NewRow(0.2, v.GlobalLabel), ui.NewRow(0.8, v.Global),
+		),
+		ui.NewRow(0.22,
+			ui.NewCol(1.0/2, ui.NewRow(0.2, v.ReqLabel), ui.NewRow(0.8, v.ReqChart)),
+			ui.NewCol(1.0/2, ui.NewRow(0.2, v.BandwidthLabel), ui.NewRow(0.8, v.BandwidthGroup)),
+		),
+		ui.NewRow(0.22,
+			ui.NewCol(1.0/2, ui.NewRow(0.2, v.ErrLabel), ui.NewRow(0.8, v.ErrGroup)),
+			ui.NewCol(1.0/2, ui.NewRow(0.2, v.HitLabel), ui.NewRow(0.8, v.HitGroup)),
+		),
+		ui.NewRow(0.22,
+			ui.NewCol(1.0/2, ui.NewRow(0.2, v.IoLabel), ui.NewRow(0.8, v.IoGroup)),
+			ui.NewCol(1.0/2, ui.NewRow(0.2, v.LoggingLabel), ui.NewRow(0.8, v.LoggingGroup)),
+		),
+	)
+}
+
+func (v *View) Resize() {
+	termWidth, termHeight := ui.TerminalDimensions()
+	v.Grid.SetRect(0, 0, termWidth, termHeight)
+}
+
+func (v *View) Render() {
+	ui.Clear()
+	ui.Render(v.Grid)
+}
+
+func (v *View) UpdateStats(block realtimeResponseData) error {
```

Comment:
This should probably be a `*realtimeResponseData`: doing it this way copies the struct as it's passed, and it has enough of a mix of pointers and values internally that it could get confusing.